modules/abi.js

```diff
@@ -11,27 +11,36 @@ function isHexType(type) {
     return _.includes(['address', 'bytes'], type) || type.match(/bytes\d+/g);
 }
 
+function padLeft(string, chars) {
+    return (new Array(chars - string.length + 1).join('0')) + string;
+};
+
 ipc.on('backendAction_decodeFunctionSignature', (event, signature, data) => {
-    var dataBuffer, paramTypes;
-    dataBuffer = new Buffer(data.slice(10, data.length), 'hex');
+    let paramTypes;
+    data = data.slice(10, data.length);
     signature = signature.match(/\((.+)\)/i);
 
     if (!signature) return;
 
     paramTypes = signature[1].split(',');
 
     try {
-        const paramsResponse = abi.rawDecode(paramTypes, dataBuffer);
-        var paramsDictArr = [];
+        const paramsResponse = abi.rawDecode(paramTypes, new Buffer(data, 'hex'));
+        let paramsDictArr = [];
 
         // Turns addresses into proper hex string
         // Turns numbers into their decimal string version
         paramTypes.forEach((type, index) => {
-            var conversionFlag = isHexType(type) ? 'hex' : null,
+            let conversionFlag = isHexType(type) ? 'hex' : null,
                 prefix = isHexType(type) ? '0x' : '';
 
             paramsResponse[index] = paramsResponse[index].toString(conversionFlag);
 
+            if(type === 'address')
+            	paramsResponse[index] = padLeft(paramsResponse[index], 40);
+            else if(type.match(/bytes\d+/g))
+            	paramsResponse[index] = padLeft(paramsResponse[index], 64);
```

Comment:
normally parameters are always in sizes of 32bytes chunks, we could also leave that out, but it makes it more consistent
Though we should add a "title" to the dapp_output in the elements package